astra/src/main/java/com/slack/astra/server/ArmeriaService.java

```diff
@@ -56,9 +57,16 @@ public static class Builder {
     private final ServerBuilder serverBuilder;
     private final List<SpanHandler> spanHandlers = new ArrayList<>();
 
-    public Builder(int port, String serviceName, PrometheusMeterRegistry prometheusMeterRegistry) {
+    private final float traceSamplingRate;
+
+    public Builder(
+        int port,
+        String serviceName,
+        PrometheusMeterRegistry prometheusMeterRegistry,
+        float traceSamplingRate) {
```

Comment:
Since this is part of the TracingConfig, I think you could drop this from the constructor and instead assign it in `withTracing`

Reply:
good point. done.